ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionLogUpdatesConversions.scala

```diff
@@ -60,26 +60,29 @@ private[events] object TransactionLogUpdatesConversions {
     )(implicit
         loggingContext: LoggingContext,
         executionContext: ExecutionContext,
-    ): ToApi[GetTransactionsResponse] = { transaction =>
-      val nonTransient = removeTransient(transaction.events)
-      val requestingParties = filter.keySet
-      Future
-        .traverse(nonTransient)(toFlatEvent(_, requestingParties, verbose, lfValueTranslation))
-        .map(flatEvents =>
-          GetTransactionsResponse(
-            Seq(
-              FlatTransaction(
-                transactionId = transaction.transactionId,
-                commandId = getCommandId(transaction.events, requestingParties),
-                workflowId = transaction.workflowId,
-                effectiveAt = Some(timestampToTimestamp(transaction.effectiveAt)),
-                events = flatEvents,
-                offset = ApiOffset.toApiString(transaction.offset),
+    ): ToApi[GetTransactionsResponse] = transaction =>
+      Future.delegate {
```

Comment:
What is the difference between Future.delegate {} and Future {} :thinking: ?

Reply:
```
Future.delegate(Future.unit) // Future[Unit]
Future(Future.unit) // Future[Future[Unit]]
```

Reply:
AFAIK, it peels off one future layer in case your {} already has a future inside i.e. Future{Future{A()}} results in Future[Future[A]], whereas Future.delegate{Future{A()}} results in Future[A]. In both cases the constructor of the nested future is invoked asynchronously